crates/bevy_pbr/src/render/pbr_lighting.wgsl

```diff
@@ -440,10 +440,93 @@ fn perceptualRoughnessToRoughness(perceptualRoughness: f32) -> f32 {
     return clampedPerceptualRoughness * clampedPerceptualRoughness;
 }
 
+// this must align with CubemapLayout in decal/clustered.rs
+const CUBEMAP_TYPE_CROSS_VERTICAL: u32 = 0;
+const CUBEMAP_TYPE_CROSS_HORIZONTAL: u32 = 1;
+const CUBEMAP_TYPE_SEQUENCE_VERTICAL: u32 = 2;
+const CUBEMAP_TYPE_SEQUENCE_HORIZONTAL: u32 = 3;
+
+const X_PLUS: u32 = 0;
+const X_MINUS: u32 = 1;
+const Y_PLUS: u32 = 2;
+const Y_MINUS: u32 = 3;
+const Z_MINUS: u32 = 4;
+const Z_PLUS: u32 = 5;
+
+fn cubemap_uv(direction: vec3<f32>, cubemap_type: u32) -> vec2<f32> {
+    let abs_direction = abs(direction);
+    let max_axis = max(abs_direction.x, max(abs_direction.y, abs_direction.z));
+
+    let face_index = select(
+        select(X_PLUS, X_MINUS, direction.x < 0.0),
+        select(
+            select(Y_PLUS, Y_MINUS, direction.y < 0.0),
+            select(Z_PLUS, Z_MINUS, direction.z < 0.0),
+            max_axis != abs_direction.y
+        ),
+        max_axis != abs_direction.x
+    );
+    
+    var face_uv: vec2<f32>;
+    var corner_uv: vec2<u32> = vec2(0, 0);
+    var face_size: vec2<f32>;
+
+    switch face_index {
+        case X_PLUS:  { face_uv = vec2<f32>(direction.z, -direction.y) / direction.x; }
+        case X_MINUS: { face_uv = vec2<f32>(-direction.z, -direction.y) / -direction.x; }
+        case Y_PLUS:  { face_uv = vec2<f32>(direction.x,  -direction.z) / direction.y; }
+        case Y_MINUS: { face_uv = vec2<f32>(direction.x, direction.z) / -direction.y; }
+        case Z_PLUS:  { face_uv = vec2<f32>(direction.x, direction.y) / direction.z; }
+        case Z_MINUS: { face_uv = vec2<f32>(direction.x, -direction.y) / -direction.z; }
+        default: {}
+    }
```

Comment:
Could avoid some divergence if you put the numerator and denominator in separate variables, and then have that funnel down to a single division operation outside the switch.

Reply:
done ... not totally convinced it helps since i don't know how good gpus are at scoping (we have a longer-lived divisor variable, vs mutually exclusive switch-scoped variables).